node/tests/SharedTests.ts

```diff
@@ -873,6 +873,60 @@ export function runBaseTests<Context>(config: {
         config.timeout,
     );
 
+    it.each([ProtocolVersion.RESP2, ProtocolVersion.RESP3])(
+        `lset with nonexisting key`,
+        async (protocol) => {
+            await runTest(async (client: BaseClient) => {
+                const key = "test_key";
+                const non_existing_key = "nonexisting";
+                const index = 0;
+                const oobIndex = 10;
+                const negativeIndex = -1;
+                const element = "zero";
+                const lpushArgs = ["four", "three", "two", "one"];
+                const expectedList = ["zero", "two", "three", "four"];
+                const expectedList2 = ["zero", "two", "three", "zero"];
+
+                // key does not exist
+                try {
+                    expect(
+                        await client.lset(non_existing_key, index, element),
+                    ).toThrow();
+                } catch (e) {
+                    expect((e as Error).message).toMatch(
+                        "An error was signalled by the server - ResponseError: no such key",
+                    );
+                }
+
+                expect(await client.lpush(key, lpushArgs)).toEqual(4);
+
+                // index out of range
+                try {
+                    expect(await client.lset(key, oobIndex, element)).toThrow();
+                } catch (e) {
+                    expect((e as Error).message).toMatch(
+                        "An error was signalled by the server - ResponseError: index out of range",
+                    );
+                }
+
+                // assert lset result
+                expect(await client.lset(key, index, element)).toEqual("OK");
+                checkSimple(await client.lrange(key, 0, negativeIndex)).toEqual(
+                    expectedList,
+                );
+
+                // assert lset with a negative index for the last element in the list
+                expect(await client.lset(key, negativeIndex, element)).toEqual(
+                    "OK",
+                );
+                checkSimple(await client.lrange(key, 0, negativeIndex)).toEqual(
+                    expectedList2,
+                );
```

Comment:
Can you add a test that attempts to call lset against a non-list key (eg a string key)?